### PR TITLE
Prometheus consensus

### PR DESCRIPTION
**Pull request by LeoHChen**
```
ubuntu@ip-172-26-12-195:~$ curl -s http://localhost:9900/metrics | grep ^hmy
hmy_consensus_bingo{consensus="bingo"} 993
hmy_consensus_bingo{consensus="signatures"} 5958
hmy_consensus_finality_bucket{le="800"} 0
hmy_consensus_finality_bucket{le="1000"} 0
hmy_consensus_finality_bucket{le="1250"} 0
hmy_consensus_finality_bucket{le="1562.5"} 8
hmy_consensus_finality_bucket{le="1953.125"} 550
hmy_consensus_finality_bucket{le="2441.40625"} 889
hmy_consensus_finality_bucket{le="3051.7578125"} 896
hmy_consensus_finality_bucket{le="3814.697265625"} 896
hmy_consensus_finality_bucket{le="4768.37158203125"} 897
hmy_consensus_finality_bucket{le="5960.4644775390625"} 897
hmy_consensus_finality_bucket{le="+Inf"} 898
hmy_consensus_finality_sum 1.736376e+06
hmy_consensus_finality_count 898
hmy_consensus_signatures{consensus="signatures"} 6
hmy_consensus_sync{consensus="in_sync"} 3
hmy_consensus_sync{consensus="out_of_sync"} 3
```

### CODE REVIEW

File: consensus/validator.go [hunk 1]
```diff
@@ -293,6 +294,13 @@ func (consensus *Consensus) onCommitted(msg *msg_pb.Message) {
 		}
 	}
 
+	ConsensusCounterVec.With(prometheus.Labels{"consensus": "bingo"}).Inc()
+	numSig := float64(consensus.NumSignaturesIncludedInBlock(blockObj))
```
Comment (rlan35) [inline consensus/validator.go:298]: can we add this counter at BINGO? It's a relatively expensive operation so we only want to do it once.

File: consensus/consensus_v2.go [hunk 2]
```diff
@@ -211,6 +212,9 @@ func (consensus *Consensus) finalCommit() {
 		Int("numTxns", len(block.Transactions())).
 		Int("numStakingTxns", len(block.StakingTransactions())).
 		Msg("HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!")
+	ConsensusCounterVec.With(prometheus.Labels{"consensus": "hooray"}).Inc()
+	ConsensusGaugeVec.With(prometheus.Labels{"consensus": "block_num"}).Set(float64(block.NumberU64()))
+	ConsensusGaugeVec.With(prometheus.Labels{"consensus": "num_commits"}).Set(float64(consensus.Decider.SignersCount(quorum.Commit)))
```
Comment (rlan35) [inline consensus/consensus_v2.go:217]: here the signers are already reset (after commitBlock). Better use the same signerCount at the beginning of finalCommit().

File: consensus/metrics.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package consensus
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
```
Comment (JackyWYX) [inline consensus/metrics.go:8]: Can we make these fields private?

File: consensus/consensus_v2.go (hunk 2 above)
Comment (LeoHChen) [reply consensus/consensus_v2.go:217]: fixed.

File: consensus/validator.go (hunk 1 above)
Comment (LeoHChen) [reply consensus/validator.go:298]: BINGO message is in node module, can't use the variables defined in consensus module.

Comment (LeoHChen) [reply consensus/validator.go:298]: let me add a func then.

File: consensus/metrics.go (hunk 3 above)
Comment (LeoHChen) [reply consensus/metrics.go:8]: done

File: consensus/validator.go (hunk 1 above)
Comment (LeoHChen) [reply consensus/validator.go]: done

**Review by rlan35 — APPROVED** [review]

**Pull request merged**